src/partition/repart.c

```diff
@@ -5005,6 +5022,10 @@ static int run(int argc, char *argv[]) {
                         return r;
         }
 
+        if (!context_verify_free_areas(context))
+                return log_error_errno(SYNTHETIC_ERRNO(ENOSPC),
+                                       "Can't grow existing partitions into available free space, refusing.");
```

Comment:
hmm, should this really be fatal?

I mean, this check will fail if PaddingMin= is larger than what is currently in effect and what space is available after the existing partition, right? I don't think we should fail in cases like this. i.e. PaddingMin= should be honoured by us for stuff we do, but if there's a violation of this constraint *already* in place before we even considered anything, then it should not matter to us.

Or am I misunderstanding what this commit does?

Reply:
PaddingMin= can also trigger the failure, but more important case is SizeMinBytes=, as reported #24553.

Reply:
But, I think in that case, we may simply set 'Partition.drop' instead of refuse the entire process.